Source/EventFlow.EventStores.EventStore/EventStoreEventPersistence.cs

```diff
@@ -122,7 +122,10 @@ private static Position ParsePosition(GlobalPosition globalPosition)
             var eventDatas = serializedEvents
                 .Select(e =>
                     {
-                        var guid = Guid.Parse(e.Metadata["guid"]);
+                        // While it might be tempting to use e.Metadata.EventId here, we can't
+                        // as EventStore won't detect optimistic concurrency exceptions then
+                        var guid = Guid.NewGuid();
```

Comment:
Do you have a way to make event ids deterministic here?  EventStore has the ability to disregard duplicate event ids - using a brand new guid every event loses the (albeit rarely needed) guarantee and event isnt written twice.

See https://eventstore.org/docs/dotnet-api/optimistic-concurrency-and-idempotence/index.html#idempotence

Reply:
What I do is use the message id's Guid and increment that per event raised.  If a command is run twice for some reason it generates a series of duplicate event ids

Reply:
The `e.Metadata.EventId` _is_ deterministic but as its currently only dependent on the aggregate ID and the event sequence number (and not the event data), there's no guarantee that it is indeed the same event. If it was possible to make 100% sure that developers implemented their events using value objects and these were implemented correctly, then it might be possible. However, if the command is executed twice its far simpler to make the executed command idempotent in the aggregate, e.g. resulting in no events if the same command is executed again. The aggregate will still be reloaded, but its easier to manage.

Agree?

Reply:
yeah if you have another way to know if the command was already run then that works great.  For me I just use the command's message id - if you have another way then :+1: